python/plugins/processing/algs/gdal/AssignProjection.py

```diff
@@ -97,3 +99,29 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         self.setOutputValue(self.OUTPUT, fileName)
 
         return commands
+
+    def postProcessAlgorithm(self, context, feedback):
+        # get output value
+        fileName = self.outputValue(self.OUTPUT)
+        if not fileName:
+            return {}
+
+        # get layer from output
+        layer = QgsProcessingUtils.mapLayerFromString(fileName, context)
```

Comment:
You shouldn't need this - rather loop through ALL layers in the project/context, and update all matching ones you find